transcom/mymove
# Amw b 17952 confirmation before converting to member expense

## [Agility ticket]
B-17952

## Summary

When rejecting a SIT extension request, the TOO should be able to convert it to "customer expense". While doing so, they should see a confirmation modal after clicking the "Convert to Customer Expense" checkbox. If they click "Yes" on the confirmation modal, it will take them back to the SIT extension review modal and the checkbox will be checked. If clicking "No" it would have taken them back to that modal but the checkbox would be unchecked.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

pre-reqs: must have a move with a SIT extension request on it
1. Access the Office side as a TOO
2. Look up the move and open it
3. Click the review button
4. If you select the Yes radio button, it will show the normal dropdown for reason categorization. If you select the No radio button, it should hide the dropdown and show a checkbox for converting to customer expense.
5. When the checkbox is visible, if you attempt to check it, it will show a confirmation modal. On that confirmation modal if you click the Yes button, it will return to the review modal and the checkbox will be checked. If you clicked the No button, it will return to the review modal and the checkbox will be unchecked.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2023-12-16 at 7 00 29 AM](https://github.com/transcom/mymove/assets/147739091/89f611c1-2137-4689-acdd-e3ed3525e4e9)
![Screenshot 2023-12-16 at 7 00 38 AM](https://github.com/transcom/mymove/assets/147739091/18ee08ae-08ce-4711-aa0d-55a00b7cddb2)
